sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObjectFactory.java

```diff
@@ -188,25 +194,32 @@ private static HandlerDecorator buildDecorator(Class<? extends HandlerDecorator>
         return decorator;
     }
 
-    private MethodInterceptor createMethodInterceptor(Class<?> sqlObjectType,
+    private InvocationHandler createInvocationHandler(Class<?> sqlObjectType,
                                                       SqlObjectConfig baseConfig,
                                                       Map<Method, Handler> handlers,
                                                       Supplier<Handle> handle) {
-        return (proxy, method, args, methodProxy) -> {
-            Handler handler = handlers.get(method);
+        return (proxy, method, args) -> {
+            if (EQUALS_METHOD.equals(method)) {
+                return proxy == args[0];
+            }
+
+            if (HASHCODE_METHOD.equals(method)) {
+                return System.identityHashCode(proxy);
+            }
 
-            // If there is no handler, pretend we are just an Object and don't open a connection (Issue #82)
-            if (handler == null) {
-                return methodProxy.invokeSuper(proxy, args);
```

Comment:
Seems like this line was already redundant, with the lazy handle stuff in place.